include/cantera/kinetics/Reaction.h

```diff
@@ -37,12 +43,39 @@ class Reaction
     //! The chemical equation for this reaction
     std::string equation() const;
 
+    //! The type of reaction
+    virtual std::string type() const = 0; // pure virtual function
+
     //! Ensure that the rate constant and other parameters for this reaction are
     //! valid.
     virtual void validate();
 
+    //! Get validity flag of reaction
+    bool valid() const {
+        return m_valid;
+    }
+
+    //! Set validity flag of reaction
+    void set_valid(bool valid) {
+        m_valid = valid;
+    }
+
+    //! Get reaction rate pointer
+    shared_ptr<RxnRate> rxnRate() {
+        return m_rate;
+    }
```

Comment:
Np about avoiding tortured abbreviations! I'd suggest `ReactionRate` with derived names `ArrheniusRate` etc. instead though as I have a pretty strong aversion to perpetuating the misnomer 'constant' more than necessary.

Reply:
Yeah, I agree that the use of the word 'constant' is not great. I was trying to avoid the redundancy of `Reaction.reactionRate`, but it's not that bad, and `ReactionRate` is a good name for the class if not the method of `Reaction`. We unfortunately can't just use `rate` since `ElementaryReaction` already has a member with that name. I guess we'll eventually be able to deprecate that and name this getter `rate`.

I'm 👍 on `FooRate` for the various types, although we may run into a little bit of trouble there since we already have a `ChebyshevRate` class.

Reply:
Interesting comment. If things proceed as anticipated (i.e. framework is adopted), `rate` and various other member variables would become part of a `ReactionRate` object. As they're variables, it's impossible to deprecate them - thoughts?

Reply:
Update: added a commit that renames member variables to `rate2` in order to pre-empt a future change.